src/components/ui/project/irysmanga/context/MangaContext.tsx

```diff
@@ -97,6 +97,31 @@ export function MangaProvider({
 		return setting;
 	}
 
+	/**
+	 * Searches the URL query params for a specific key and returns the
+	 * value pointed to it as a number.
+	 * @param key Key of the search param.
+	 * @param def Default value to return if the specified search key is not found.
+	 * @returns The value parsed as a number if found, else def.
+	 */
+	function getNumberFromQuery(key: string, def: number): number {
```

Comment:
nit/suggestion: I guess it might be cleaner to just implicitly do keys rather than explicit ones as query params? Usually I see it like `base/chapter_number/page_number` on places like guya or MD.

Reply:
I thought about that but it looks like whatever routing system the website uses doesn't really like doing it the `blahblahblah.com/chapter_number/page_number` way (it 404s). We can ask @GoldElysium if the routing can be updated to support this, but in the meantime I implemented it like this because it was the path of least resistance

Reply:
gotcha, sounds fine to me then.

Reply:
Also, regarding routing, no, since it'd break everything else. An alternative that might look a bit cleaner could be to put the route in the fragment/hash instead. E.g. `.../projects/route#1/3` or `...#chapter1/3`.